src/contracts/StakingChef.sol

```diff
@@ -7,82 +7,64 @@ import "./Chef.sol";
 
 contract StakingChef is Chef {
     using SafeMath for uint256;
-    Staker[] private _stakes;
 
-    //mapping(address => mapping(address => userStakes)) public userStakes;
+    mapping(address => Staker) public staker;
 
     constructor(address metricTokenAddress) {
         setMetricToken(metricTokenAddress);
         setMetricPerBlock(4);
         toggleRewards(false);
     }
-// --------------------------------------------------------------------- staking functions
-    function stakeMetric(
-        address newAddress,
-        uint256 metricAmount,
-        uint256 newStartDate
-    ) external nonDuplicated(newAddress) {
-    if (areRewardsActive() && getTotalAllocationShares() > 0) {
+
+    // --------------------------------------------------------------------- staking functions
+    function stakeMetric(uint256 metricAmount, uint256 newStartDate) external nonDuplicated(msg.sender) {
+        if (areRewardsActive() && getTotalAllocationShares() > 0) {
             updateAccumulatedStakingRewards();
         }
-        Staker memory stake = Staker({
-            stakeAddress: newAddress,
-            metricAmount: metricAmount, 
+        staker[msg.sender] = Staker({
+            stakeAddress: msg.sender,
+            metricAmount: metricAmount,
             startDate: newStartDate,
             rewardDebt: metricAmount.mul(getLifetimeShareValue()).div(ACC_METRIC_PRECISION),
             claimable: 0
         });
 
-        _stakes.push(stake);
-        addTotalAllocShares(stake.metricAmount);
-        SafeERC20.safeTransferFrom(IERC20(getMetricToken()), msg.sender, address(this), stake.metricAmount); 
+        addTotalAllocShares(staker[msg.sender].metricAmount);
+        SafeERC20.safeTransferFrom(IERC20(getMetricToken()), msg.sender, address(this), staker[msg.sender].metricAmount);
     }
 
-    function updateStaker(
-        address stakeAddress,
-        uint256 stakeIndex,
-        uint256 metricAmount 
-    ) public {
+    function updateStaker(uint256 metricAmount) public {
         if (areRewardsActive() && getTotalAllocationShares() > 0) {
             updateAccumulatedStakingRewards();
         }
-        addTotalAllocShares(_stakes[stakeIndex].metricAmount, metricAmount);
-        _stakes[stakeIndex].stakeAddress = stakeAddress;
-        _stakes[stakeIndex].metricAmount = metricAmount;
+        addTotalAllocShares(staker[msg.sender].metricAmount, metricAmount);
+        staker[msg.sender].metricAmount = metricAmount;
     }
 
-    function removeStaker(uint256 stakeIndex) external {
-        require(stakeIndex < _stakes.length, "Index more than stakes length");
+    function removeStaker() external {
```

Comment:
I guess I'm having trouble distinguishing removeStaker from withdraw Principal in theory shouldn't they do the same thing?

Reply:
I'm thinking both should be combined and be called unStakeMetric

Reply:
honestly we can revisit all the External API's when ya build out the integration with the site right?  We don't have to get it perfect before it's used